public/apps/configuration/panels/role-edit.tsx

```diff
@@ -47,24 +58,191 @@ const TITLE_TEXT_DICT = {
   create: 'Create Role',
   edit: 'Edit Role',
   duplicate: 'Duplicate Role',
-}
+};
 
 type OptionSeletion = EuiComboBoxOptionOption[];
 
-function buildPermissionOptions(optionsList: string[]) {
-  return optionsList.map(e => ({ label: e }));
+interface RoleIndexPermissionStateClass {
+  indexPatterns: OptionSeletion;
+  dls: string;
+  flsMethod: 'exclude' | 'include';
+  fls: OptionSeletion;
+  maskedFields: OptionSeletion;
+  allowedActions: OptionSeletion;
+}
+
+function getEmptyIndexPermission(): RoleIndexPermissionStateClass {
+  return {
+    indexPatterns: [],
+    dls: '',
+    flsMethod: 'exclude',
+    fls: [],
+    maskedFields: [],
+    allowedActions: [],
+  };
+}
+
+function buildIndexPermissionState(
+  indexPerm: RoleIndexPermission[]
+): RoleIndexPermissionStateClass[] {
+  return indexPerm.map(perm => ({
+    indexPatterns: perm.index_patterns.map(stringToComboBoxOption),
+    dls: perm.dls,
+    // Leading ~ indicates exclude.
+    flsMethod: perm.fls.some((s: string) => !s.startsWith('~')) ? 'exclude' : 'include',
+    fls: perm.fls.map((s: string) => s.replace(/^~/, '')).map(stringToComboBoxOption),
+    maskedFields: [],
+    allowedActions: [],
+  }));
+}
+
+const FIELD_LEVEL_SECURITY_PLACEHOLDER = `{
+    "bool": {
+        "must": {
+            "match": {
+                "genres": "Comedy"
```

Comment:
would quite understand why use "genres": "Comedy" as example security placeholder, if it is not intented, would you revise to better one in next PR?

Reply:
It is copied from the UI mockup. Tech writer may revise it.